src/services/member/plugins/profile/schemas.ts

```diff
@@ -1,66 +1,70 @@
-import { S } from 'fluent-json-schema';
+import { Type } from '@sinclair/typebox';
 import { StatusCodes } from 'http-status-codes';
 
 import { FastifySchema } from 'fastify';
 
-import { error, uuid } from '../../../../schemas/fluent-schema';
-import { partialMember } from '../../../item/schema';
+import { customType, registerSchemaAsRef } from '../../../../plugins/typebox';
+import { errorSchemaRef } from '../../../../schemas/global';
 
-const memberSharedSchema = S.object()
-  .additionalProperties(false)
-  .prop('bio', S.string())
-  .prop('facebookID', S.anyOf([S.string(), S.null()]))
-  .prop('linkedinID', S.anyOf([S.string(), S.null()]))
-  .prop('twitterID', S.anyOf([S.string(), S.null()]));
+const inputProfileMemberSchemaRef = registerSchemaAsRef(
+  'memberShared',
+  'Member Shared',
+  Type.Object(
+    {
+      bio: Type.String(),
+      facebookID: Type.String(),
+      linkedinID: Type.String(),
+      twitterID: Type.String(),
+      visibility: Type.Boolean(),
+    },
+    { additionalProperties: false },
+  ),
+);
 
-export const profileMember = S.object()
-  .prop('id', uuid)
-  .prop('bio', S.string())
-  .prop('visibility', S.boolean())
-  .prop('facebookID', S.string())
-  .prop('linkedinID', S.string())
-  .prop('twitterID', S.string())
-  .prop('createdAt', S.string().format('date-time'))
-  .prop('updatedAt', S.string().format('date-time'))
-  .prop('member', partialMember);
+export const profileMemberSchemaRef = registerSchemaAsRef(
+  'profileMember',
+  'Profile Member',
+  Type.Object({
+    id: customType.UUID(),
+    bio: Type.String(),
+    facebookID: Type.String(),
+    linkedinID: Type.String(),
+    twitterID: Type.String(),
+    visibility: Type.Boolean(),
+    createdAt: customType.DateTime(),
+    updatedAt: customType.DateTime(),
+  }),
+);
 
 export const createProfile = {
-  body: S.object()
-    .additionalProperties(false)
-    .prop('visibility', S.boolean())
-    // QUESTION: why is the bio always required ? Can't we only update a social link ?
-    .required(['bio'])
-    .extend(memberSharedSchema),
+  body: inputProfileMemberSchemaRef,
   response: {
-    [StatusCodes.CREATED]: profileMember,
-    [StatusCodes.UNAUTHORIZED]: error,
+    [StatusCodes.CREATED]: profileMemberSchemaRef,
+    [StatusCodes.UNAUTHORIZED]: errorSchemaRef,
   },
-};
+} as const satisfies FastifySchema;
 
-export const getProfileForMember: FastifySchema = {
-  params: S.object().additionalProperties(false).prop('memberId', uuid),
+export const getProfileForMember = {
+  params: Type.Object({ memberId: customType.UUID() }, { additionalProperties: false }),
   response: {
-    [StatusCodes.OK]: profileMember,
-    [StatusCodes.NO_CONTENT]: S.null(),
-    [StatusCodes.UNAUTHORIZED]: error,
+    [StatusCodes.OK]: profileMemberSchemaRef,
+    [StatusCodes.NO_CONTENT]: Type.Null(),
```

Comment:
I think we should annotate why this uses a 204 instead of a 404. I think it has something to do with how the query-client reacts to 4xx codes. They are treated as errors, but we want it to be a null return with success. Not sure this is good pattern since I was asking myself if this should not have been a 404 code.